searchlib/src/tests/grouping/grouping_test.cpp

```diff
@@ -1859,4 +1860,24 @@ TEST(GroupingTest, test_that_attributes_can_be_unconditionally_converted_to_docu
     EXPECT_TRUE(attrRequest.getRoot().getAggregationResult(0).getExpression()->inherits(DocumentFieldNode::classId));
 }
 
+TEST(GroupingTest, test_bad_grouping)
+{
+    Grouping baseRequest;
+    baseRequest.setRoot(Group().addChild(Group().setId(FloatBucketResultNode(0.2, 0.3)))
```

Comment:
Consider adding a comment about _why_ this is a bad grouping request (i.e. non strictly increasing bucket order)